invenio_curations/config.py

```diff
@@ -31,6 +31,24 @@
 }
 """Invenio requests facets."""
 
+CURATIONS_PERMISSIONS_VIA_GRANTS = True
+"""Share access to records by creating access grants for records under review.
+
+This eliminates the requirement for overriding the record permission policy,
+but introduces additional artifacts in the system for each record.
+"""
+
+CURATIONS_ALLOW_PUBLISHING_EDITS = False
+"""Allow publishing of metadata edits for already published records.
+
+This allows users to modify their record's metadata and publish those changes
+without going through another curation review.
+The idea here is that the record has already passed through an initial quality check
+anyway and it's very unlikely that anybody would update the metadata afterwards in
+a way that decreases their quality.
+Even if that happens, InvenioRDM stores `revisions` for record metadata.
```

Comment:
That sounds great. How can we retrieve the metadata of past revisions?

Reply:
through `record.revisions`:
```python
In [5]: len(list(RDMRecord.pid.resolve("d48y6-qhj80").revisions))
Out[5]: 2

In [6]: list(RDMRecord.pid.resolve("d48y6-qhj80").revisions)[0]
Out[6]: 
{'id': 'd48y6-qhj80',
 'pid': {'pk': 133805, 'status': 'R', 'obj_type': 'rec', 'pid_type': 'recid'},
 'pids': {'doi': {'client': 'datacite',
   'provider': 'datacite',
   'identifier': '10.70124/d48y6-qhj80'},
  'oai': {'provider': 'oai',
   'identifier': 'oai:test.researchdata.tuwien.ac.at:d48y6-qhj80'}},
 'files': {'enabled': True},
 'access': {'files': 'restricted',
  'record': 'public',
  'embargo': {'until': None, 'active': False, 'reason': None}},
 '$schema': 'local://records/record-v6.0.0.json',
 'metadata': {'title': 'E-LAUTE A-Wn Mus.Hs. 18688 n59 33v audio',
  'rights': [{'id': 'cc-by-4.0'}],
  'version': '1.0.0',
  'creators': [{'role': {'id': 'contactperson'},
    'affiliations': [{'id': '04d836q62'}],
    'person_or_org': {'name': 'Jaklin, Julia Maria',
     'type': 'personal',
     'given_name': 'Julia Maria',
     'family_name': 'Jaklin'}}],
  'publisher': 'TU Wien',
  'description': '<p>A recording of a lute piece located in the manuscript A-Wn Mus.Hs. 18688 n59 33v. More information on the source in RISM: <a href="https://opac.rism.info/rism/Record/rism600141880" target="_blank" rel="noopener">https://opac.rism.info/rism/Record/rism600141880</a>.&nbsp;<br>A scan of the score can be found in the &Ouml;NB: <a href="https://digital.onb.ac.at/RepViewer/viewer.faces?doc=DTL_4439424&amp;order=71&amp;view=SINGLE" target="_blank" rel="noopener">https://digital.onb.ac.at/RepViewer/viewer.faces?doc=DTL_4439424&amp;order=71&amp;view=SINGLE</a>.</p>',
  'resource_type': {'id': 'sound'},
  'publication_date': '2024-09-09'},
 'media_files': {'enabled': False},
 'custom_fields': {}}
```

from what i remember you can even get a diff between versions pretty easily, but that's a story for another day.

:warning: note that invenio only stores revisions for *records*, but not *drafts*!
from what i remember, the main reasoning was that this would very likely start filling up the DB with data that's not crazy valuable (though it could simplify the diff detection in the curations service component, and maybe even help the reviewer by displaying a diff).
perhaps the draft revisions could be deleted when the record gets published?

Reply:
Ah yeah that is awesome. Having a diff for reviewers would certainly help.

For drafts, your approach does sound good as well. We would just have to fetch the revision when the request was submitted last and diff it with the current revision. Sounds like a future feature :smile: